src/Entity/User.php

```diff
@@ -1231,6 +1231,11 @@ public function hasValidLinkToRelay(Centre $relay): bool
         return null !== $userRelay && $userRelay->getBValid();
     }
 
+    public function hasPermissionOnRelay(Centre $relay, string $permission): bool
+    {
+        return $this->isMembre() && $this->getUserRelay($relay)?->hasDroit($permission);
```

Comment:
Ici, tu ne peux pas faire un `$this->isGranted($permission, $relay)` ? Et coder ton voter en fonction ?